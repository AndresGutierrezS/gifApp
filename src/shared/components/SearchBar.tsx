import { useEffect, useState } from "react";

interface Props {
    placeHolder?: string;

    onQuery: (query: string) => void;
}

export const SearchBar = ({placeHolder = 'Search', onQuery}: Props) => {
  const [query, setQuery] = useState('');

  useEffect(() => {
    const timeoutId = setTimeout(() => {
      onQuery(query)
    }, 700);
    return () => {
      clearTimeout(timeoutId)
    }
  }, [query, onQuery])

  const handleSearch = () => {
    onQuery(query)
  }

  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      handleSearch();
    }
  }  
  return (
    <div className="search-container">
        <input 
          type="text" 
          placeholder={placeHolder}
          value={query}
          onChange={(event) => setQuery(event.target.value)}
          onKeyDown={handleKeyDown}
        />
        <button onClick={handleSearch}>Search</button>
    </div>
  )
}
