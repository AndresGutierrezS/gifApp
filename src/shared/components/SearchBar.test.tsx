import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import { SearchBar } from "./SearchBar";


describe('SearchBar', () => {
    test('Should render SearchBar correctly', () => {
        const { container } = render(<SearchBar onQuery={() => {}}/>);

        expect(container).toMatchSnapshot();
    });

    test('Should call onQuery with the correct value after 700ms', async () => {
        const onQuery = vi.fn();
        render(<SearchBar onQuery={onQuery}/>);

        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { value: 'test' }});

        await waitFor(() => {
            expect(onQuery).toHaveBeenCalled();
            expect(onQuery).toHaveBeenCalledWith('test');
        })
    });
});