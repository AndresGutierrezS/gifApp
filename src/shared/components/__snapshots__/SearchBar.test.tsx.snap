// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > Should render SearchBar correctly 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      placeholder="Search"
      type="text"
      value=""
    />
    <button>
      Search
    </button>
  </div>
</div>
`;
